docs/explanation/bind-charm-architecture.md

```diff
@@ -0,0 +1,78 @@
+# Bind charm architecture
+
+The following diagram shows the different modules of this charm
+and how they interact with each other.
+
+```mermaid
+C4Component
+title Bind Operator Components
```

Comment:
I think in this section, I was expecting something more like this:
https://github.com/canonical/github-runner-operator/blob/main/docs/explanation/charm-architecture.md
Instead of file names more like components/modules and external integrations if any.
wdyt

Reply:
I thought a bit about this and I prefer filenames since they directly indicate where to look for things.  
I could add the module names but sometimes there is no class (for events.py for example).

By external integrations, you mean imports of external libraries ? Because this charm's code doesn't discuss with the outside world except through its relations

Reply:
I discussed this case with Erin as well, and since we don't have a specific pattern to follow and all the approaches seem reasonable, we can work on standardizing the architecture in the future.

External I mean components/other charms.